tikv-client-common/src/errors.rs

```diff
@@ -28,6 +28,12 @@ pub enum Error {
     /// An operation requires a primary key, but the transaction was empty.
     #[error("transaction has no primary key")]
     NoPrimaryKey,
+    #[error(
+        "The operation does is not supported in raw-atomic mode, please consider using an atomic client"
+    )]
+    UnsupportedInAtomicMode,
+    #[error("The operation is only supported in raw-atomic mode, please consider using a non-atomic raw client")]
+    UnsupportedInNonAtomicMode,
```

Comment:
I think just one UnsupportedMode error would be better.

Reply:
OK. I'm also thinking that the error handling does not provide a good experience. It's often hard to determine the concrete error from the error message. Maybe we should attach the trace with the error or find some other ways to improve it.